### ISSUE TITLE
Exceptions in a step could be handled better

### ISSUE THREAD

**Issue by magnars** [flosell/lambdacd#179]
When an exception happens in a step, lambdacd shows only 

> No output.

Maybe it would be useful to include the error message and the stack trace at this point? I guess this would be the responsibility of the runner?

**Comment by magnars** [comment]
I would be happy to contribute something here, if you could point me in the right direction. Or maybe the best solution is a separate project that instruments a pipeline somehow?

**Comment by magnars** [comment]
I just had the opposite happen - the output did in fact show the exception. So now I don't understand when the exception is shown and when not. Possibly something to do with dev vs prod?

**Comment by flosell** [comment]
Normally, exceptions (as in, everything inheriting from `Exception`) thrown in a step should appear in the step-output including the stack trace. 

[This](https://github.com/flosell/lambdacd/blob/62e2870/src/clj/lambdacd/execution/internal/execute_step.clj#L26-L33) is the code responsible for it and at least on first glance it looks reasonable to me... 

The only thing I could imagine would be that your step threw a `Throwable` that wasn't an `Exception` or that the exception came from somewhere in the LambdaCD internals that happen after the error handling.

**Comment by magnars** [comment]
Yes, that looks entirely reasonable - and does indeed work in my current pipeline. I'll try to reconstruct the situation, I think I remember what went wrong.

**Comment by flosell** [comment]
@magnars were you able to reproduce the problem? Let me know if you need help with this. 
If the problem isn't reproducible I'd close the issue for now until we find more information about it.
